tracing-opentelemetry/src/layer.rs

```diff
@@ -804,8 +804,12 @@ where
     /// [`ERROR`]: tracing::Level::ERROR
     /// [`Error`]: opentelemetry::trace::StatusCode::Error
     fn on_event(&self, event: &Event<'_>, ctx: Context<'_, S>) {
-        // Ignore events that are not in the context of a span
-        if let Some(span) = ctx.lookup_current() {
+        // Ignore events that have no explicit parent set *and* are not in the context of a span
+        if let Some(span) = event
+            .parent()
+            .and_then(|id| ctx.span(id))
+            .or_else(|| ctx.lookup_current())
```

Comment:
nit, take it or leave it: the `Context` type has an [`event_span` method](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/layer/struct.Context.html#method.event_span) which essentially performs this logic, so this could be simplified to:
```suggestion
        if let Some(span) = ctx.event_span(event) {
```